mmdet3d/datasets/transforms/transforms_3d.py

```diff
@@ -2352,3 +2354,134 @@ def transform(self, input_dict: dict) -> dict:
             if len(input_dict[key]) == 0:
                 input_dict.pop(key)
         return input_dict
+
+
+@TRANSFORMS.register_module()
+class PolarMix(BaseTransform):
+    """PolarMix data augmentation.
+
+    Required Keys:
+
+    - points (:obj:`BasePoints`)
+    - pts_semantic_mask (np.int64)
+    - mix_results (List[dict])
+
+    Modified Keys:
+
+    - points (:obj:`BasePoints`)
+    - pts_semantic_mask (np.int64)
+
+    Args:
+        instance_classes (List[int]): Semantic masks which represent the
+            instance.
+        swap_ratio (float): Swap ratio of two point cloud. Defaults to 0.5.
+        rotate_paste_ratio (float): Rotate paste ratio. Defaults to 1.0.
+    """
+
+    def __init__(self,
+                 instance_classes: List[int],
+                 swap_ratio: float = 0.5,
+                 rotate_paste_ratio: float = 1.0) -> None:
+        assert is_list_of(instance_classes, int)
+        self.instance_classes = instance_classes
+        self.swap_ratio = swap_ratio
+        self.rotate_paste_ratio = rotate_paste_ratio
+
+    def get_indexes(self, dataset: BaseDataset) -> int:
+        """Call function to collect indexes.
+
+        Args:
+            dataset (:obj:`BaseDataset`): The dataset.
+
+        Returns:
+            int: Index.
+        """
+        index = random.randint(0, len(dataset))
+        return index
+
+    def transform(self, input_dict: dict) -> dict:
+        """PolarMix transform function.
+
+        Args:
+            input_dict (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: output dict after transformtaion
+        """
+
+        assert 'mix_results' in input_dict
```

Comment:
I don't understand where the `mix_results` is generated

Reply:
It seems that we need to implement a new dataset_wrapper following [ MultiImageMixDataset](https://github.com/open-mmlab/mmdetection/blob/dev-3.x/mmdet/datasets/dataset_wrappers.py#L12) to use `PolarMix`.

Reply:
So should we add a new dataset wrapper in this PR?

Reply:
I will add the `dataset wrapper` in another PR.

Reply:
I have tried with `MultiImageMixDataset`, and we can directly use MultiImageMixDataset without adding new dataset wrapper.

Reply:
We should follow the implementation in MMYOLO, which do not need MultiImageMixDataset.
MultiImageMixDataset is a bit slower

Reply:
Updated, whether should we also implement a cached version?

Reply:
unnecessary for now